chatops/test_hubot.bats

```diff
@@ -0,0 +1,76 @@
+
+load 'test_helpers/bats-support/load'
+load 'test_helpers/bats-assert/load'
+
+
+@test "StackStorm's client connection" {
+	run st2 action execute core.local cmd=echo
+	assert_success
+	assert_line --index 0          "To get the results, execute:"
+	assert_line --index 1 --regexp "^ st2 execution get [0-9a-f]*$"
+	assert_line --index 2          "To view output in real-time, execute:"
+	assert_line --index 3 --regexp "^ st2 execution tail [0-9a-f]*$"
+}
+
+@test "npm directory exists" {
+	run eval "(cd /opt/stackstorm/chatops; npm list | grep hubot-stackstorm)"
+	assert_success
+	assert_output --partial "hubot-stackstorm@"
+}
+
+@test "some StackStorm aliases are enabled" {
+	run st2 action-alias list -a enabled -j
+	assert_success
+	assert_output --partial '"enabled": true'
+}
+
+@test "chatops.notify rule exists" {
+	run st2 rule list -p chatops -j
+	assert_success
+	assert_output --partial '"ref": "chatops.notify"'
+	assert_output --partial '"enabled": true'
+}
+
+@test "hubot help command works" {
+	run eval "{ echo -n; sleep 5; echo 'hubot help'; echo; sleep 5; } | bin/hubot --test"
+	assert_success
+	assert_output --partial '!help - Displays all of the help commands'
+	assert_output --regexp '[[:digit:]]{1,} commands are loaded'
+}
+
+@test "chatops.post_message execution and receive status works" {
+	RANDOM_CHANNEL_NAME=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
```

Comment:
We had some issues with random channel names in the past, but I assume you tested it and it works (IIRC, bot needs to be invited to the channel so it works, and this didn't work with random channels so we've used a fixed channel for tests).

Reply:
This doesn't actually post to any Slack channels - this bypasses Slack entirely by posting the message directly to hubot/st2chatops, as if the message was posted through Slack and prefixed with `!`. The tests in `chatops/test_hubot.bats` are not end-to-end tests; they simply test that the st2chatops service under test communicates with other st2 services under test.

Here's the part of the chain that this tests (Slack isn't involved):

```
hubot/st2chatops
  -> st2
-> hubot/st2chatops
```

Here's the full chain tested by the end-to-end tests:

```
user (or a bot that impersonates a user)
  -> Slack
    -> hubot/st2chatops (which is just another Slackbot)
      -> st2
    -> hubot/st2chatops
  -> Slack
-> user (or the bot that is impersonating a user)
```

Reply:
Thanks for clarifying.